src/shell/atuin.bash

```diff
@@ -34,6 +34,6 @@ fi
 
 if [[ -z ${ATUIN_NOBIND} ]]; then
     bind -x '"\C-r": __atuin_history'
-    bind -x '"\e[A": __atuin_history'
-    bind -x '"\eOA": __atuin_history'
+    bind -x '"\e[A": __atuin_history --filter-mode session'
+    bind -x '"\eOA": __atuin_history --filter-mode session'
```

Comment:
On second thought, an alternative may be to have an additional configuration setting for filter mode when interactive mode is invoked via _up_ key bindings.

Reply:
Not sure which would be the best way to tell atuin that it invoked from the _up_ key binding between:
* adding a `--shell-up-key-binding` CLI argument:
  ```suggestion
    bind -x '"\eOA": __atuin_history --shell-up-key-binding'
  ```
* defining a `ATUIN_SHELL_UP_KEY_BINDING` environment variable:
```suggestion
    bind -x '"\eOA": ATUIN_SHELL_UP_KEY_BINDING=1 __atuin_history'
```

Reply:
I'd rather configure this in our config file, and globally for all shells (not just bash). I'd also be pretty happy for `--shell-up-key-binding` to be how it's invoked!

> I have a slight preference for hardcoding --filter-mode session in the init scripts, since up is for looking for a previous command and ctrl-r is for searching the history.

I think generally I'd agree with you there @patricksjackson, but I'm also a little cautious to change defaults at this point without a strong opinion for why. Currently I mostly feel lukewarm on the topic

Reply:
Done.